src/contracts/version/Registry.ts

```diff
@@ -67,12 +70,19 @@ export class Registry extends Contract {
   }
 
   public registerExchangeAdapter(from: Address, args: ExchangeAdapterInformation) {
-    return this.createTransaction('registerExchangeAdapter', from, [
-      args.exchange,
-      args.adapter,
-      args.takesCustody,
-      args.sigs.map(sig => hexToBytes(utf8ToHex(sig))),
-    ]);
+    return this.createTransaction(
```

Comment:
The method arguments for `createTransaction` are becoming a bit unwieldy, what do you think? Should we maybe switch to an object instead? 

```
this.createTransaction({
  method: 'registerExchangeAdapter',
  from,
  validate,
  ...  
});
```